### PR TITLE
executor: fix unstable test TestStaleReadPrepare

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43269

close https://github.com/pingcap/tidb/issues/33331

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
check_dev2

**Review by xhebox — APPROVED** [review]

File: executor/stale_txn_test.go [hunk 1]
```diff
@@ -1045,7 +1045,9 @@ func TestStaleReadPrepare(t *testing.T) {
 	tk.MustExec("set @a=now(6)")
 	time.Sleep(5 * time.Millisecond)
 	tk.MustExec("update t1 set v=100 where id=1")
-	tk.MustQuery("select * from t1").Check(testkit.Rows("1 100"))
+	tk.EventuallyMustQueryAndCheck(
+		"select * from t1",
+		nil, testkit.Rows("1 100"), 100*time.Millisecond, 5*time.Millisecond)
 	tk.MustExec("prepare s1 from 'select * from t1 as of timestamp @a where id=1'")
 	tk.MustQuery("execute s1").Check(testkit.Rows("1 10"))
```
Comment (Defined2014) [inline executor/stale_txn_test.go:1052]: The issue is this line? 

```
[2022-03-23T02:52:07.128Z]         	            	expected: "[1 10]\n"
```

Comment (Defined2014) [reply executor/stale_txn_test.go:1052]: maybe we need add a sleep between L1044 - L1045

Comment (Defined2014) [reply executor/stale_txn_test.go:1052]: or use `tidb_current_ts` instead of `now`

**Review by xhebox — APPROVED** [review]

File: executor/stale_txn_test.go [hunk 2]
```diff
@@ -1042,7 +1042,7 @@ func TestStaleReadPrepare(t *testing.T) {
 
 	tk.MustExec("create table t1 (id int, v int)")
 	tk.MustExec("insert into t1 values (1,10)")
-	tk.MustExec("set @a=now(6)")
+	tk.MustExec("set @a=@@tidb_current_ts")
```
Comment (Defined2014) [inline executor/stale_txn_test.go:1045]:
```suggestion
    time.Sleep(time.Millisecond)
    tk.MustExec("begin")
	tk.MustExec("set @a=tidb_parse_tso(@@tidb_current_ts)")
	tk.MustExec("commit")
```

File: executor/stale_txn_test.go [hunk 3]
```diff
@@ -1042,7 +1042,7 @@ func TestStaleReadPrepare(t *testing.T) {
...
 	time.Sleep(5 * time.Millisecond)
```
Comment (Defined2014) [inline executor/stale_txn_test.go:1046]:
```suggestion
	time.Sleep(3 * time.Millisecond)
```

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**